envoy_client/models.py

```diff
@@ -126,56 +240,178 @@ class PowerSourceType(enum.IntEnum):
+    6 = Other generation system
+    80 = Other storage system
+    81 = Electric vehicle
+    82 = EVSE
+    83 = Combined PV and storage
+    All other values reserved.
+    """
+    na_unknown = 0
+    virtual_or_mixed_DER = 1
+    reciprocating_engine = 2
+    fuel_cell = 3
+    pv_system = 4
+    combined_heat_power = 5
+    other_generation = 6
+    other_storage = 80
+    electric_vehicle = 81
+    EVSE = 82
+    combined_pv_storage = 83
 
 
 class DERCapability(BaseModel):
+    """
+    The DER resource exposes the capabilities of a specific distributed energy resource, 
+    referred to as its nameplate ratings. Ratings are read-only values established by the 
+    DER manufacturer by design or manufactured configuration, for instance, the continuous 
+    delivered active power rating capability in watts (rtgMaxW), and are available by reading 
+    the DERCapability resource.
+    """
     modes_supported: int = Field(default=1, alias='modesSupported')
-    rtg_max_w: float = Field(alias='rtgMaxW')
-    type_: int = Field(alias='type')
+    rtg_max_a: Optional[ValueWithMultiplier] = Field(alias='rtgMaxA')
+    rtg_max_ah: Optional[ValueWithMultiplier] = Field(alias='rtgMaxAh')
+    rtg_max_w: Optional[ValueWithMultiplier] = Field(alias='rtgMaxW')
+    rtg_max_charge_rate_va: Optional[ValueWithMultiplier] = Field(alias='rtgMaxChargeRateVA')
+    rtg_max_charge_rate_w: Optional[ValueWithMultiplier] = Field(alias='rtgMaxChargeRateW')
+    rtg_max_discharge_rate_va: Optional[ValueWithMultiplier] = Field(alias='rtgMaxDischargeRateVA')
+    rtg_max_discharge_rate_w: Optional[ValueWithMultiplier] = Field(alias='rtgMaxDischargeRateW')
+    type_: DERType = Field(alias='type')
+
+    class XmlTemplate:
+        create = {
+            'by_alias': True,
+            'exclude_none': True
+        }
 
 
 class DER(BaseModel):
-    der_capability: DERCapability = Field(alias='DERCapability')
+    """Contains links to DER resources
+    """
+    der_capability: Optional[DERCapability] = Field(alias='DERCapability')
+    class XmlTemplate:
+        create = {
+            'include': {},
+            'by_alias': True,
+        }
+        link = {}
+        show = {}
+
 
+class ConnectionPoint(BaseModel):
+    """Extension object containing information about the connection point to which 
+    a device is connected. Under normal circumstances, the `meter_id` attribute
+    is used to convey the NMI at the device premise to the utility server.
+    
+    The `connection_point_id` is normally not used by the aggregator client, as it
+    references IDs normally used internally by the DNSP.
+    """
+    connection_point_id: Optional[str] = Field(alias='connectionPointID')
+    meter_id: Optional[str] = Field(alias='meterID')
+
+    class XmlTemplate:
+        create = {
+            'include': {'connection_point_id', 'meter_id'},
+            'by_alias': True
+        }
+        show = create
+        list = create
 
 class EndDevice(BaseModel):
```

Comment:
Maybe these container classes should be renamed so they're distinct from the 2030 objects - to avoid confusion